engine/access/rpc/backend/script_executor.go

```diff
@@ -21,25 +22,41 @@ type ScriptExecutor struct {
 	// initialized is used to signal that the index and executor are ready
 	initialized *atomic.Bool
 
-	// init is used to ensure that the object is initialized only once
-	init sync.Once
+	// minCompatibleHeight and maxCompatibleHeight are used to limit the block range that can be queried using local execution
+	// to ensure only blocks that are compatible with the node's current software version are allowed.
+	// Note: this is a temporary solution for cadence/fvm upgrades while version beacon support is added
+	minCompatibleHeight *atomic.Uint64
+	maxCompatibleHeight *atomic.Uint64
 }
 
 func NewScriptExecutor() *ScriptExecutor {
 	return &ScriptExecutor{
-		initialized: atomic.NewBool(false),
+		initialized:         atomic.NewBool(false),
+		minCompatibleHeight: atomic.NewUint64(0),
+		maxCompatibleHeight: atomic.NewUint64(math.MaxUint64),
 	}
 }
 
+// SetMinCompatibleHeight sets the lowest block height (inclusive) that can be queried using local execution
+// Use this to limit the executable block range supported by the node's current software version.
+func (s *ScriptExecutor) SetMinCompatibleHeight(height uint64) {
+	s.minCompatibleHeight.Store(height)
```

Comment:
better to log the change of the height range?

Reply:
addressed in [2134775](https://github.com/onflow/flow-go/pull/5283/commits/21347755c21a534794ddf2a73d7d54c7bac559b6)